packages/uniforms/src/BaseForm.tsx

```diff
@@ -251,6 +255,7 @@ export class BaseForm<
       event.preventDefault();
       event.stopPropagation();
     }
+    this.setState({ submitted: true });
```

Comment:
It'd be better to skip unneeded updates:
```suggestion

    this.setState(state => state.submitted ? null : { submitted: true });
```

Reply:
react already does that internally.

Reply:
No, it's not - here's a demo [CodeSandbox](https://codesandbox.io/s/practical-gareth-di8sb?file=/src/App.js). In class components you have to return `null`, in `useState` hook it works only if `newState === oldState`.